hudi-common/src/main/java/org/apache/hudi/common/util/CommitUtils.java

```diff
@@ -84,7 +86,10 @@ public static HoodieCommitMetadata buildMetadata(List<HoodieWriteStat> writeStat
     if (extraMetadata.isPresent()) {
       extraMetadata.get().forEach(commitMetadata::addMetadata);
     }
-    commitMetadata.addMetadata(HoodieCommitMetadata.SCHEMA_KEY, schemaToStoreInCommit == null ? "" : schemaToStoreInCommit);
+    // NULL Schema should not be written to commit metadata
+    if (!(schemaToStoreInCommit != null && schemaToStoreInCommit.equals(NULL_SCHEMA_STR))) {
```

Comment:
Should we keep the behavior consistent? Always add the schema as empty string if ether the schema is null or its NULL_SCHEMA_STR?

Reply:
Fixed.